zbus/src/connection/mod.rs

```diff
@@ -1005,35 +1005,13 @@ impl Connection {
                                     }
                                 }
                             }
-                            let member = match hdr.member() {
-                                Some(member) => member,
-                                None => {
-                                    warn!("Got a method call with no `MEMBER` field: {}", msg);
-
-                                    continue;
-                                }
-                            };
-                            trace!("Got `{}`. Will spawn a task for dispatch..", msg);
-                            let executor = conn.inner.executor.clone();
-                            let task_name = format!("`{member}` method dispatcher");
-                            // SAFETY: msg becomes 'static when moved to the task, and dropped after
-                            let hdr = unsafe { std::mem::transmute::<_, Header<'static>>(hdr) };
-                            executor
-                                .spawn(
-                                    async move {
-                                        trace!("spawned a task to dispatch `{}`.", msg);
-                                        let server = conn.object_server();
-                                        if let Err(e) = server.dispatch_call(&msg, &hdr).await {
-                                            debug!(
-                                                "Error dispatching message. Message: {:?}, error: {:?}",
-                                                msg, e
-                                            );
-                                        }
-                                    }
-                                    .instrument(trace_span!("{}", task_name)),
-                                    &task_name,
-                                )
-                                .detach();
```

Comment:
This is very confusing. So until the commit "♻️ zb: drop the mandatory task spawning", we were essentially spawning tasks twice for each method? So if we only reverted this commit, we'll be in a weird state.

Why not first do the needed refactor so that you don't need this additional commit?

Reply:
Well, if we drop spawning before introducing it, we are in a weird state (worse since it regresses). What's wrong with spawning twice? Alternatively, just squash all changes, but that's harder to review/bisect.

Reply:
whatever, I don't have the energy for debate this one. :)